Feature: Royalty fee and local marketplace listing fees update

chexware opened the pull request.

justinphamnz commented on pallets/auction/src/lib.rs:
```diff
@@ -956,22 +958,20 @@ pub mod pallet {
+			match listing_level {
+				ListingLevel::Local(metaverse_id) => {
+					class_fund = T::MetaverseInfoSource::get_metaverse_treasury(metaverse_id);
+					royalty_fee =
```
royalty_fee and listing fee are 2 different concepts. Royalty fee will go to NFT fund, listing fee will go to metaverse treasury.

chexware replied:
sure will update the code to support both

chexware replied:
@justinphamnz  do the global marketplace have a listing fee as well?

justinphamnz reviewed (changes requested):
Good work @chexware. Listing fee info should be included inside Auction struct when new auction created. It will prevent listing fee changes after auction started

chexware commented:
@justinphamnz  I've updated the auction items to have fee collection property so the metaverse owner can't change their listing fee during the auction. Also, when an auction is finalized booth royalty fee and listing fee are collected. Let me know if I need to change anything else

chexware commented:
@justinphamnz I've updated the auction item type to include listing fee property so metaverse owner can't change listing fee after an auction/buy_now is created. Also updated the code so both listing fee and royalty fees are collected when the auction is finalized. Let me know if I need to make any other changes.
